### ISSUE TITLE
:source-map-include-sources-content false generates invalid source maps

### ISSUE THREAD

**Issue by mikekap** [thheller/shadow-cljs#785]
I have a repo with a repro at https://github.com/mikekap/shadow-cljs-repro . I used `npx shadow-cljs release --debug main && npx shadow-cljs server` to see the output below in the chrome console.

With the source content option on, the exceptions that look like:
```
$repro$ns1$throwit$$    	@	ns1.cljs:6
$repro$ns1$i_am_an_fn_2$$    	@	ns1.cljs:16
$repro$main$bounce$$    	@	main.cljs:9
(anonymous)                	@	main.cljs:17
setInterval (async)		
$repro$main$run$$        	@	main.cljs:14
(anonymous)             	@	append.js:2
(anonymous)             	@	append.js:2
```

Without the source content option, they look like:
```
$repro$ns1$throwit$$    	@	ns1.cljs:4
$repro$ns1$i_am_an_fn_2$$	@	ns1.cljs:17
$repro$main$bounce$$     	@	main.cljs:4
(anonymous)              	@	main.cljs:11
setInterval (async)		
$repro$main$run$$        	@	main.cljs:7
(anonymous)             	@	append.js:2
(anonymous)              	@	append.js:2
```

The line numbers are definitely wrong, with no real pattern that's easy to correct. I'm not sure if the problem extends to filenames, though we haven't seen that as much in our codebase.

**Comment by thheller** [comment]
Why are you using this in the first place?

I can't change how Chrome (or whatever) handles source maps when the sources are missing. shadow-cljs does not write them since this is more of an internal option for testing. The generated source maps are identical so I'll put that down to the missing sources.

**Comment by mikekap** [comment]
Hey thanks for the quick response. We're using this to keep our source code private. Our use case hasn't changed since this feature was created: https://github.com/thheller/shadow-cljs/issues/434#issuecomment-461107938 . If that's not a supported use case anymore, how could we exclude our source code from our source map output?

Digging around, the source of the mismapping seems to be some code around https://github.com/thheller/shadow-cljs/blob/master/src/main/shadow/build/closure.clj#L915-L917 . Removing the `source-map-include-sources-content` check there produces valid source maps for me again. Would you accept this change and a test to make sure this keeps working in the future?

**Comment by thheller** [comment]
Oops that was indeed incorrect. Fixed in `2.11.4`.
